lib/OpenQA/Task/Job/Limit.pm

```diff
@@ -19,22 +19,34 @@ use Mojo::Base 'Mojolicious::Plugin';
 use File::Spec::Functions 'catfile';
 use File::Basename qw(basename dirname);
 use OpenQA::Utils qw(:DEFAULT imagesdir);
+use Scalar::Util 'looks_like_number';
+use List::Util 'min';
+
+# define default parameter for batch processing
+use constant SCREENSHOTS_PER_BATCH            => 200000;       # number of screenshots processed in one DB query
+use constant AVERAGE_EXECUTION_TIME_PER_BATCH => 8;            # estimation for SCREENSHOTS_PER_BATCH
+use constant DESIRED_MINION_JOB_RUNTIME       => (60 * 60);    # aim for one hour per Minion job
+use constant BATCHES_PER_MINION_JOB => (DESIRED_MINION_JOB_RUNTIME / AVERAGE_EXECUTION_TIME_PER_BATCH);
 
 sub register {
     my ($self, $app) = @_;
-    $app->minion->add_task(limit_results_and_logs => sub { _limit($app, @_) });
+    my $minion = $app->minion;
+    $minion->add_task(limit_results_and_logs => sub { _limit($app, @_) });
+    $minion->add_task(limit_screenshots      => sub { _limit_screenshots($app, @_) });
```

Comment:
`$app` is also available as `$job->app`. There is no need to capture it in the closure.
```suggestion
    $minion->add_task(limit_results_and_logs => \&_limit);
    $minion->add_task(limit_screenshots      => \& _limit_screenshots);
```